lxc/completion.go

```diff
@@ -211,6 +212,21 @@ func (g *cmdGlobal) cmpImages(toComplete string) ([]string, cobra.ShellCompDirec
 }
 
 func (g *cmdGlobal) cmpInstanceAllKeys(instanceName string) ([]string, cobra.ShellCompDirective) {
+	var keys []string
+	cmpDirectives := cobra.ShellCompDirectiveNoFileComp
+
+	if strings.Contains(instanceName, ":") {
+		instanceName = strings.Split(instanceName, ":")[1]
```

Comment:
You can likely use `strings.Cut` here to get the instance name and check if the colon is present in one step.